### PR TITLE
Fix: enable input but don't be allowed to submit when agent init

### PR DESCRIPTION
**Pull request by iFurySt**
it seems a regression. i also processed to prevent the Enter event can't to submitting the messages.

https://github.com/OpenDevin/OpenDevin/assets/16201837/264f326b-f521-4202-bb45-ed6de64b5c42

### CODE REVIEW

File: frontend/src/components/ChatInput.tsx [hunk 1]
```diff
@@ -25,6 +24,7 @@ function ChatInput({ disabled, onSendMessage }: ChatInputProps) {
   };
 
   const onKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (disabled) return;
     if (event.key === "Enter" && !event.shiftKey && !isComposing) {
       event.preventDefault(); // prevent a new line
       handleSendChatMessage();
```
Comment (yimothysu) [inline frontend/src/components/ChatInput.tsx:30]:
```suggestion
    if (event.key === "Enter" && !event.shiftKey && !isComposing) {
      event.preventDefault(); // prevent a new line
      if (!disabled) {
          handleSendChatMessage();
      }
```

Currently, we prevent `Enter` from adding a newline if composing or the shift key is pressed. IMO this should happen regardless of whether the input is disabled, for consistency.

**Review by yimothysu — APPROVED** [review]

**Comment by rbren** [comment]
Seems to work well! Only complaint is that if I insert linebreaks into the message, they seem to get squished down to a single line in the chat log. Probably a separate issue

**Comment by iFurySt** [comment]
> Seems to work well! Only complaint is that if I insert linebreaks into the message, they seem to get squished down to a single line in the chat log. Probably a separate issue

can you screenshot and open a new issue? frontend folks can try to fix it 😄

File: frontend/src/components/ChatInput.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Textarea } from "@nextui-org/react";
 import React from "react";
 import { useTranslation } from "react-i18next";
-import { VscArrowUp } from "react-icons/vsc";
+import { VscSend, VscArrowUp } from "react-icons/vsc";
```
Comment (rbren) [inline frontend/src/components/ChatInput.tsx:4]:
```suggestion
import { VscArrowUp } from "react-icons/vsc";
```

**Pull request merged**